k3s-io/k3s
# allow coredns override extensions

#### Proposed Changes ####

Allow CoreDNS override extensions in default Corefile

Other K8s implementations such as AKS[1] and DigitalOcean[2] have similar configuration

This would enable ConfigMaps to adjust the CoreDNS settings without creating a whole copy of coredns.yaml.

1: https://learn.microsoft.com/en-us/azure/aks/coredns-custom
2: https://docs.digitalocean.com/products/kubernetes/how-to/customize-coredns/

#### Types of Changes ####

New Feature

#### Verification ####

1. Patch Corefile with `import /etc/coredns/custom/*.override` in the `.:53` default server block
2. Create a `coredns-custom` ConfigMap with some custom settings for the default server block, e.g. `rewrite name substring svc.example.com svc.cluster.local`
3. Restart CoreDNS pod
4. Test that the custom config has taken effect. For my example, run `nslookup my-pod.my-svc.my-namespace.svc.example.com` or `getent hosts ...` in a pod and confirm it resolves to the same name as `my-pod.my-svc.my-namespace.svc.cluster.local`

#### Testing ####




#### Linked Issues ####

* https://github.com/k3s-io/k3s/issues/7639

#### User-Facing Change ####

```release-note

```

#### Further Comments ####
